src/main/java/org/opensearch/ad/transport/RCFResultTransportAction.java

```diff
@@ -57,13 +55,18 @@ protected void doExecute(Task task, RCFResultRequest request, ActionListener<RCF
             listener.onFailure(new LimitExceededException(request.getAdID(), CommonErrorMessages.MEMORY_CIRCUIT_BROKEN_ERR_MSG));
             return;
         }
+        /* @anomaly.detection Commented until we have extension support for hashring : https://github.com/opensearch-project/opensearch-sdk-java/issues/200 
         Optional<DiscoveryNode> remoteNode = hashRing.getNodeByAddress(request.remoteAddress());
         if (!remoteNode.isPresent()) {
             listener.onFailure(new ConnectException("Can't find remote node by address"));
             return;
         }
         String remoteNodeId = remoteNode.get().getId();
         Version remoteAdVersion = hashRing.getAdVersion(remoteNodeId);
+        */
+        DiscoveryNode remoteNode = extensionsRunner.getExtensionNode();
+        String remoteNodeId = remoteNode.getId();
```

Comment:
Where are we using `remoteNodeId` here?

Reply:
I kept this to reduce the diff but I can change it